File: src/Agent/Agent.Web/Client/src/src/Common/Contracts/DataPlane/ReadFileResult.ts
/**
 * Structured result from a file read operation for rich UI rendering.
 */
export interface ReadFileResult {
    /** Relative path to the file from sandbox root */
    filePath: string;
    /** 1-based start line number */
    startLine: number;
    /** 1-based end line number (inclusive) */
    endLine: number;
    /** Total number of lines in the file */
    totalLines: number;
    /** The file content (lines joined with \n) */
    content: string;
    /** Whether the content was truncated due to line limit */
    isTruncated: boolean;
    /** Error message if the read failed */
    error?: string;
}

File: src/Agent/Agent.Core/Models/Api/v1/ReadFileResult.cs
// ------------------------------------------------------------
//  Copyright (c) Microsoft Corporation.  All rights reserved.
// ------------------------------------------------------------

namespace Agent.Core.Models.Api.v1;

/// <summary>
/// Structured result from a file read operation for rich UI rendering.
/// </summary>
public class ReadFileResult
{
    /// <summary>
    /// Relative path to the file from sandbox root.
    /// </summary>
    public string FilePath { get; set; } = string.Empty;

    /// <summary>
    /// 1-based start line number.
    /// </summary>
    public int StartLine { get; set; }

    /// <summary>
    /// 1-based end line number (inclusive).
    /// </summary>
    public int EndLine { get; set; }

    /// <summary>
    /// Total number of lines in the file.
    /// </summary>
    public int TotalLines { get; set; }

    /// <summary>
    /// The file content (lines joined with \n).
    /// </summary>
    public string Content { get; set; } = string.Empty;

    /// <summary>
    /// Whether the content was truncated due to line limit.
    /// </summary>
    public bool IsTruncated { get; set; }

    /// <summary>
    /// Error message if the read failed.
    /// </summary>
    public string? Error { get; set; }
}
